aiohttp_client_cache/backends/dynamodb.py

```diff
@@ -1,148 +1,179 @@
-                    {'AttributeName': 'key', 'KeyType': 'RANGE'},
-                ],
-                ProvisionedThroughput={
-                    'ReadCapacityUnits': read_capacity_units,
-                    'WriteCapacityUnits': write_capacity_units,
-                },
-            )
-        except ClientError:
-            pass
-
-        self._table = self.connection.Table(table_name)
-        self._table.wait_until_exists()
-
-    def _scan_table(self) -> Dict:
-        return self._table.query(
-            ExpressionAttributeValues={':Namespace': self.namespace},
-            ExpressionAttributeNames={'#N': 'namespace'},
-            KeyConditionExpression='#N = :Namespace',
-        )
-
-    @staticmethod
-    def unpickle(response_item: Dict) -> ResponseOrKey:
-        return BaseCache.unpickle((response_item or {}).get('value'))
+        self.key_attr_name = key_attr_name
+        self.val_attr_name = val_attr_name
+        self.create_if_not_exists = create_if_not_exists
+        self.context = context
+        self._table = None
+
+    async def get_table(self):
+        if not self._table:
+            # Re-use the service resource if it's already been created
+            if self.context.cls:
+                conn = self.context.cls
+            # otherwise create
+            else:
+                # should we try to call aexit later if we auto enter here?
+                conn = await self.context.__aenter__()
+
+            self._table = await conn.Table(self.table_name)
+            if self.create_if_not_exists:
+                try:
+                    await conn.create_table(
+                        AttributeDefinitions=[
+                            {
+                                'AttributeName': self.key_attr_name,
+                                'AttributeType': 'S',
+                            },
+                        ],
+                        TableName=self.table_name,
+                        KeySchema=[
+                            {
+                                'AttributeName': self.key_attr_name,
+                                'KeyType': 'HASH',
+                            },
+                        ],
+                        BillingMode="PAY_PER_REQUEST",
+                    )
+                    await self._table.wait_until_exists()
+                except ClientError as e:
+                    if e.response["Error"]["Code"] != "ResourceInUseException":
+                        raise
+
+        return self._table
+
+    def _doc(self, key) -> Dict:
+        return {self.key_attr_name: f'{self.namespace}:{key}'}
+
+    async def delete(self, key: str) -> None:
+        doc = self._doc(key)
+        table = await self.get_table()
+        await table.delete_item(Key=doc)
+
+    async def read(self, key: str) -> ResponseOrKey:
+        table = await self.get_table()
+        response = await table.get_item(Key=self._doc(key), ProjectionExpression=self.val_attr_name)
```

Comment:
Projecting here in anticipation of adding extra attrs. A TTL impl could override write to include an additional "ttl" attr but otherwise wouldn't need to change much else, other than maybe skipping delete and of course defining ttl on the table during provisioning.

Reply:
Sounds good to me.